File: leetcode/transpose-matrix.java
//https://leetcode.com/problems/transpose-matrix/

class Solution {
    public int[][] transpose(int[][] matrix) {
        int n = matrix[0].length;
        int m = matrix.length;
        int[][] transpose = new int[n][m];
        
        for(int i = 0; i<m; i++){
            for(int j = 0; j<n; j++){
                transpose[j][i] = matrix[i][j];
            }
        }
        
        return transpose;
    }
}

File: leetcode/transpose_matrix.py
# https://leetcode.com/problems/transpose-matrix


class Solution:
    def transpose(self, matrix: list[list[int]]) -> list[list[int]]:
        updated_matrix: list[list[int]] = [[[0] for _ in range(len(matrix))] for _ in range(len(matrix[0]))]

        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                updated_matrix[j][i] = matrix[i][j]

        return updated_matrix
